### DISCUSSION TITLE
Modeling SCM: Aqueous and Calcite

### DISCUSSION THREAD

**Discussion by JMCarmona10** [reaktoro/reaktoro#370] [Q&A]
### Description

In Reaktoro API reference,  there are two options to track the reactions in an aqueous and calcite interface. Equilibrium (for species) and kinetics ( for dissolution-minerals). I made a simple code just to see if i can track the reactions in the following figure using reaktoro:  
![calcite](https://github.com/reaktoro/reaktoro/assets/116467897/f3ca5b84-4474-4c76-a7f9-7dc41545a331)
with the following reactions: 
![reactions__1](https://github.com/reaktoro/reaktoro/assets/116467897/f146242e-1477-4aed-b409-679d86f8cb26)

But I encountered an issue when using Reaktoro. The EquilibriumSolver allowed me to track the formation of aqueous species and the decreasing concentration of reactants. However, calcite dissolution occurred instantaneously. Hoping to control the timing of dissolution to re-use the same calcite to run another simulation using different concentrations of species (brine), I switched to KineticsSolver, specifying parameters for calcite: volume, surface area, and rate. This allowed me to regulate the timing of calcite dissolution, but I could no longer view or track any secondary aqueous reactions. 

Is there a way to see the reactions in an aqueous solution by time tracking while controlling the calcite dissolution (on the surface) simultaneously? Or i am wrong trying to see both results at the same time. 

Here is my simple code: 
[Simple_code.zip](https://github.com/reaktoro/reaktoro/files/14427249/Simple_code.zip)

### What is your operating system?

None

**Comment by allanleal** [comment]
Hi @JMCarmona10 - in your script, I see the following line:

~~~py
mineral = MineralPhase("Calcite >CO3- >CaOH >CaOH2+ >CaSO4- >CaCO3- >CO3H >CO3Ca+ >CO3Mg+")
~~~

This means a solid phase with all the listed components. I think what you want is:

~~~py
calcite = MineralPhase("Calcite")
surface = MineralPhase(">CO3- >CaOH >CaOH2+ >CaSO4- >CaCO3- >CO3H >CO3Ca+ >CO3Mg+")

system = ChemicalSystem(db, solution, calcite, surface)
~~~

Note the `surface` phase above is a solution phase with activities of end-members computed as their mole fractions.

**Important:** The log(K) values you are proving are not for a formation reaction (at least in some cases). For example, for the formation reaction `H+ + OH- = H2O`, at 298.15 K, its log(K) should be 13.99, not -13.99. Please revise them.

Go step by step - start with an equilibrium problem containing only H2O, H+, OH-. And then gradually add more species. This will allow you to identify errors more easily for your custom database.
